lib/Doctrine/Common/DataFixtures/AbstractFixture.php

```diff
@@ -71,13 +71,18 @@ public function addReference($name, $object)
      *
      * @see Doctrine\Common\DataFixtures\ReferenceRepository::getReference
      *
-     * @param string $name
+     * @param string      $name
+     * @param string|null $class
      *
      * @return object
+     *
+     * @template T of object
+     * @psalm-param class-string<T>|null $class
+     * @psalm-return $class is null ? object : T
      */
-    public function getReference($name)
+    public function getReference($name, $class = null) // NEXT_MAJOR: Make $class mandatory
```

Comment:
it cannot be made mandatory in next major if you don't report a deprecation when it is omitted.

Reply:
I added a deprecation if it's ok this way.